app/src/main/java/com/google/android/ground/ui/offlineareas/selector/OfflineAreaSelectorFragment.kt

```diff
@@ -68,8 +71,17 @@ class OfflineAreaSelectorFragment : AbstractMapContainerFragment() {
 
   override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
     super.onViewCreated(view, savedInstanceState)
-    viewModel.isDownloadProgressVisible.observe(viewLifecycleOwner) {
-      showDownloadProgressDialog(it)
+    viewLifecycleOwner.lifecycleScope.launch {
+      repeatOnLifecycle(Lifecycle.State.RESUMED) {
```

Comment:
Is this necessary? Can it be removed given that coroutines aren't getting suspended.

Reply:
ah, those are live data, fixed now